agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/telemetry/TelemetryChannel.java

```diff
@@ -83,28 +84,39 @@
   private final HttpPipeline pipeline;
   private final URL endpointUrl;
   private final LocalFileWriter localFileWriter;
+  private final LocalFileLoader localFileLoader;
 
   public static TelemetryChannel create(
       URL endpointUrl,
       LocalFileWriter localFileWriter,
+      LocalFileLoader localFileLoader,
       Configuration.AadAuthentication aadAuthentication) {
     HttpPipeline httpPipeline = LazyHttpClient.newHttpPipeLine(aadAuthentication, true);
-    return new TelemetryChannel(httpPipeline, endpointUrl, localFileWriter);
+    return new TelemetryChannel(httpPipeline, endpointUrl, localFileWriter, localFileLoader);
   }
 
-  public static TelemetryChannel create(URL endpointUrl, LocalFileWriter localFileWriter) {
-    return create(endpointUrl, localFileWriter, null);
+  public static TelemetryChannel create(
+      URL endpointUrl, LocalFileWriter localFileWriter, LocalFileLoader localFileLoader) {
+    return create(endpointUrl, localFileWriter, localFileLoader, null);
   }
 
   public CompletableResultCode sendRawBytes(ByteBuffer buffer) {
-    return internalSend(singletonList(buffer), null);
+    CompletableResultCode resultCode = internalSend(singletonList(buffer), null);
+    localFileLoader.updateProcessedFileStatus(resultCode.isSuccess());
```

Comment:
I think it would be clearer to move calling the `localFileLoader` up to the caller of this method (LocalFileSender) which is also where we already have a reference to the local file loader, and then no need to inject LocalFileLoader into TelemetryChannel